Git-commit: 22914d33ca33ad2d7857eba85fb3063fcbd53c0b
From: Matthias Reichl <hias@horus.com>
Date: Sun, 11 Oct 2020 00:48:55 +0200
Subject: [PATCH] bcm2835-dma: only reserve channel 0 if legacy dma driver is
 enabled
Patch-mainline: Queued in openEuler repo, version 5.10.0-4.5.0
References: openEuler-21.03
Git-repo: https://gitee.com/openeuler/kernel.git

raspberrypi inclusion
category: feature
bugzilla: 50432

--------------------------------

If CONFIG_DMA_BCM2708 isn't enabled there's no need to mask out
one of the already scarce DMA channels.

Signed-off-by: Matthias Reichl <hias@horus.com>
Signed-off-by: Fang Yafen <yafen@iscas.ac.cn>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Kai Liu <kai.liu@suse.com>
---
 drivers/dma/bcm2835-dma.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/dma/bcm2835-dma.c b/drivers/dma/bcm2835-dma.c
index d1127b9058a4..0cbfa9d559a2 100644
--- a/drivers/dma/bcm2835-dma.c
+++ b/drivers/dma/bcm2835-dma.c
@@ -1283,6 +1283,7 @@ static int bcm2835_dma_probe(struct platform_device *pdev)
 		goto err_no_dma;
 	}
 
+#ifdef CONFIG_DMA_BCM2708
 	/* One channel is reserved for the legacy API */
 	if (chans_available & BCM2835_DMA_BULK_MASK) {
 		rc = bcm_dmaman_probe(pdev, base,
@@ -1293,6 +1294,7 @@ static int bcm2835_dma_probe(struct platform_device *pdev)
 
 		chans_available &= ~BCM2835_DMA_BULK_MASK;
 	}
+#endif
 
 	/* And possibly one for the 40-bit DMA memcpy API */
 	if (chans_available & od->cfg_data->chan_40bit_mask &
-- 
2.31.1

